WIP[KHWEBA-19] add tags

Vladyslav-Plakhuta opened the pull request:
![screenshot](https://user-images.githubusercontent.com/34074565/37559740-b1ee2bf6-2a33-11e8-9db5-dc4c301f606b.png)

Vladyslav-Plakhuta committed:
insert task addTag

gift-a commented on src/components/app/app.js:
```diff
@@ -4,8 +4,14 @@ import { DrawerComponent } from "../drawer/drawer";
 import { AboutComponent } from "../about-dialog/about-dialog";
+import { PieChartComponent } from "../pie-chart/pie-chart";
+import { BarChartComponent } from "../bar-chart/bar-chart";
+import { TableTransactionsComponent } from "../table-transactions/table-transactions";
```
charts and table are in transactions component

gift-a commented on src/components/app/app.js:
```diff
@@ -4,8 +4,14 @@ import { DrawerComponent } from "../drawer/drawer";
+import { BarChartComponent } from "../bar-chart/bar-chart";
+import { TableTransactionsComponent } from "../table-transactions/table-transactions";
+import { StorageService } from "../../services/storage";
+import { Tag } from "../../model/tag";
```
we decided not to use models

gift-a commented on src/components/app/app.js:
```diff
@@ -4,8 +4,14 @@ import { DrawerComponent } from "../drawer/drawer";
+import { PieChartComponent } from "../pie-chart/pie-chart";
+import { BarChartComponent } from "../bar-chart/bar-chart";
+import { TableTransactionsComponent } from "../table-transactions/table-transactions";
+import { StorageService } from "../../services/storage";
```
you need your own storage service

gift-a commented on src/components/app/app.js:
```diff
@@ -35,23 +41,44 @@ export class AppComponent {
+  mountTransactionsComponent(tags) {
+    this.transactionsComponent = new TransactionsComponent(
+      this.transactionsMountPoint,
+      { tags: tags }
```
what does this method do? You don't need to do anything with transaction component. your sphere of work is Drawer. TransactionsComponent is a content part. 
And by using this method you rewrite `this.transactionsComponent` from row 45

gift-a commented on src/components/drawer/drawer.js:
```diff
@@ -15,38 +17,42 @@ export class DrawerComponent {
-    this.moreBtnMountPoints = this.mountPoint.querySelectorAll(
-      ".drawer__more-button"
+    this.tagsMountPoint = this.mountPoint.querySelector(
+      ".drawer__tags-mountpoint"
```
maybe `.drawer__tags-mount-point`?

Vladyslav-Plakhuta committed:
remove unnecessary import from app.js

ihor7016 commented on src/components/app/app.js:
```diff
@@ -35,23 +38,35 @@ export class AppComponent {
   }
 
+  getTags() {
+    StorageService.get("tags").then(tags => {
```
you should not use storage service directly -- but make your wrapper around it to handle just tags

ihor7016 replied:
moreover -- why do you need this here? you are supposed to get your tags in `tags` component

ihor7016 commented on src/model/tag.js:
```diff
@@ -0,0 +1,5 @@
+export class Tag {
```
please remove it -- we do not use models

ihor7016 commented on src/index.scss:
```diff
@@ -28,3 +28,4 @@
 @import "./components/button-more/button-more";
 @import "./components/table-transactions/table-transactions";
 @import "./components/chart/chart";
+@import "./components/tags/tags-component/tags-component";
```
we do not use `component` in file name -- so it should be called just `tags`

ihor7016 commented on src/components/drawer/drawer.js:
```diff
@@ -15,38 +17,42 @@ export class DrawerComponent {
-        onDeleteClicked: this.handleDeleteClick.bind(this),
-        onEditClicked: this.handleEditClick.bind(this)
-      }).mount();
+  initTags(tags) {
```
please move this logic of working with tags to tags component

ihor7016 commented on src/components/drawer/drawer.js:
```diff
@@ -15,38 +17,42 @@ export class DrawerComponent {
+  }
+
+  onAddTagConfirmed(tag) {
+    this.addTag(tag);
```
again these handlers for adding/removing tags should be in tags component

ihor7016 commented on src/components/tags/tags-component/tags-component.js:
```diff
@@ -0,0 +1,90 @@
+  }
+
+  handleAddTagConfirmed(tag) {
+    TagService.get().then(tags => {
```
please move this logic to `TagService` to method called `add`

ihor7016 commented on src/components/tags/tags-component/tags-component.js:
```diff
@@ -0,0 +1,90 @@
+    let element = document.createElement("div");
+    element.innerHTML = tagItemTemplate({ tag: tag });
+    let childNode = this.tagList.childNodes[2];
+    this.tagList.insertBefore(element.firstChild, childNode);
```
better

```js
this.tagList.innerHTML = tagItemTemplate({ tag: tag }) + this.tagList.innerHTML
```

ihor7016 commented:
![image](https://user-images.githubusercontent.com/36405386/37568023-29f7b0f0-2ad8-11e8-9534-1752bce85945.png)

please store tags as just list of tag names like this
```js
['tag1', 'tag2', 'tag3']
```

Vladyslav-Plakhuta committed:
fixed remarks

Vladyslav-Plakhuta committed:
Merge branch 'master' into KHWEBA-19

gift-a commented on src/components/app/app.js:
```diff
@@ -3,7 +3,8 @@ import template from "./app.html";
 import { ToolbarComponent } from "../toolbar/toolbar";
 import { TransactionsComponent } from "../transactions/transactions";
-import { AddTagComponent } from "../add-tag-dialog/add-tag-dialog";
+import { AboutComponent } from "../about-dialog/about-dialog";
```
it is in toolbar

gift-a commented on src/components/app/app.js:
```diff
@@ -18,9 +19,10 @@ export class AppComponent {
+    this.addAccountMountPoint = this.mountPoint.querySelector(
+      ".app__add-account-dialog"
     );
+    this.aboutMountPoint = this.mountPoint.querySelector(".app__about-dialog");
```
it is in toolbar

gift-a commented on src/components/app/app.js:
```diff
@@ -29,34 +31,37 @@ export class AppComponent {
       }
     );
     this.transactionsComponent.mount();
+    this.aboutDialog = new AboutComponent(this.aboutMountPoint);
```
it is in toolbar

gift-a commented on src/components/app/app.js:
```diff
@@ -29,34 +31,37 @@ export class AppComponent {
 
-  handleAddTagOnclick() {
-    this.addTagDialog.showDialog();
+  handleAboutOnclick() {
```
it is in toolbar

gift-a commented on src/components/drawer/drawer.js:
```diff
@@ -35,31 +38,15 @@ export class DrawerComponent {
-
   addEventListeners() {
-    this.addTagButton.addEventListener(
+    this.addAccountButton.addEventListener(
```
why add account?

gift-a commented on src/components/drawer/drawer.js:
```diff
@@ -35,31 +38,15 @@ export class DrawerComponent {
+    this.addAccountButton.addEventListener(
       "click",
-      this.handleAddTagOnclick.bind(this)
+      this.handleAddAccountClick.bind(this)
```
why add account?

gift-a commented on src/components/drawer/drawer.js:
```diff
@@ -35,31 +38,15 @@ export class DrawerComponent {
 
-  handleAddTagOnclick() {
-    this.props.onAddTagClick();
+  handleAddAccountClick() {
```
why add account?

gift-a commented on src/components/tags/tags.js:
```diff
@@ -0,0 +1,108 @@
+    });
+  }
+
+  initAddTagDialogComponent() {
```
it should name `mountChildren`

Vladyslav-Plakhuta commented:
removed 'model', made wrapper around storage service, moved logic of working with tags to 'tags component', renamed files in 'tags component'.

review pls

gift-a commented on package.json:
```diff
@@ -52,6 +52,7 @@
     "chart.js": "^2.7.2",
-    "normalize.css": "^8.0.0"
+    "normalize.css": "^8.0.0",
+    "npm": "^5.7.1"
```
why do you need `npm` here?

gift-a commented on src/index.scss:
```diff
@@ -28,6 +28,4 @@
 @import "./components/chart/chart";
-@import "./components/accounts/accounts";
-@import "./components/custom-select/custom-select";
-@import "./components/about-dialog/about-dialog";
```
why do you delete this? you break app

gift-a commented on src/components/drawer/drawer.js:
```diff
@@ -69,9 +56,8 @@ export class DrawerComponent {
-    this.initMoreBtns();
     this.initMDC();
     this.addEventListeners();
-    this.initAccountComponent();
```
why do you delete this? it is not yours

gift-a commented on src/components/app/app.js:
```diff
@@ -29,34 +31,37 @@ export class AppComponent {
     this.drawerComponent.mount();
   }
 
-  handleTransactionAdded(data) {
```
why do you delete this? it is not yours

gift-a commented on src/components/app/app.html:
```diff
@@ -5,6 +5,6 @@
     <div class="app__container-toolbar-point"></div>
     <div class="app__container-content"></div>
-    <div class="app__add-tag-dialog"></div>
+    <div class="app__add-account-dialog"></div>
```
why account?
